src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisReferralsPerformanceReportProcessor.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.reporting.ndmis.performance
+
+import mu.KLogging
+import net.logstash.logback.argument.StructuredArguments
+import org.springframework.batch.item.ItemProcessor
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.ActionPlanService
+
+@Component
+class NdmisReferralsPerformanceReportProcessor(
+  private val actionPlanService: ActionPlanService,
+) : ItemProcessor<Referral, ReferralsData> {
+  companion object : KLogging()
+
+  override fun process(referral: Referral): ReferralsData {
+    logger.debug("processing referral {}", StructuredArguments.kv("referralId", referral.id))
+
+    if (referral.sentAt == null) throw RuntimeException("invalid referral passed to report processor; referral has not been sent")
```

Comment:
it's not ideal to see this code repeated in each processor - wonder if there's a way we can abstract this assertion?